test/OpenTelemetry.Tests/Metrics/InMemoryExporterTests.cs

```diff
@@ -45,9 +47,9 @@ public void InMemoryExporterShouldDeepCopyMetricPoints()
             meterProvider.ForceFlush();
 
             var metric = exportedItems[0]; // Only one Metric object is added to the collection at this point
-            var metricPointsEnumerator = metric.GetMetricPoints().GetEnumerator();
+            var metricPointsEnumerator = metric.MetricPoints.GetEnumerator();
             Assert.True(metricPointsEnumerator.MoveNext()); // One MetricPoint is emitted for the Metric
-            ref readonly var metricPointForFirstExport = ref metricPointsEnumerator.Current;
```

Comment:
We should let the `ref` and `ref readonly` semantics be here as we want to check that that `MetricPoint` in the `ICollection` is not updated.

Reply:
with this change, `MetricPoints` is a `IReadOnlyList<MetricPoint>`. It's enumerator cannot be returned by reference:

>Error	CS8156	An expression cannot be used in this context because it may not be passed or returned by reference

Reply:
In that case, we should enumerate over the first MetricSnapshot's `MetricPoints` again to confirm that it has not changed otherwise we are testing it on the copy that was created before the second snapshot was added.

Reply:
Is there any reason to keep using Enumerators here?
I think we can accomplish the same thing just interacting with the list.

Reply:
Not anymore now that you have changed the type to a list in `MetricSnapshot`.